From 0bffe9fd5a8ecb557cc8987b69f9ee63014e39aa Mon Sep 17 00:00:00 2001
From: "wenbiao.zhang" <wenbiao.zhang@amlogic.com>
Date: Sat, 18 Jan 2014 17:09:00 +0800
Subject: [PATCH 3077/5965] PD #86264: not set registers the first time when
 tvmode is the same to avoid blue screen for cvbs

---
 drivers/amlogic/display/vout/tvconf.c  | 7 ++++++-
 drivers/amlogic/display/vout/tvconf.h  | 2 ++
 drivers/amlogic/display/vout/tvoutc.c  | 9 +++++++++
 drivers/amlogic/hdmi/hdmi_tx/hdmi_tx.c | 6 ++++--
 4 files changed, 21 insertions(+), 3 deletions(-)

diff --git a/drivers/amlogic/display/vout/tvconf.c b/drivers/amlogic/display/vout/tvconf.c
index 41aa7e727f14..86062e12c741 100755
--- a/drivers/amlogic/display/vout/tvconf.c
+++ b/drivers/amlogic/display/vout/tvconf.c
@@ -366,6 +366,11 @@ static const vinfo_t *tv_get_current_info(void)
 	return info->vinfo;
 }
 
+tvmode_t vmode_to_tvmode(vmode_t mod) 
+{
+    return vmode_tvmode_tab[mod];
+}
+
 static int tv_set_current_vmode(vmode_t mod)
 {
 	if ((mod&VMODE_MODE_BIT_MASK)> VMODE_SXGA)
@@ -377,7 +382,7 @@ static int tv_set_current_vmode(vmode_t mod)
 	switch_vpu_mem_pd_vmod(info->vinfo->mode, VPU_MEM_POWER_ON);
 	request_vpu_clk_vmod(info->vinfo->video_clk, info->vinfo->mode);
 #endif
-	tvoutc_setmode(vmode_tvmode_tab[mod]);
+	tvoutc_setmode(vmode_to_tvmode(mod));
 //	change_vdac_setting(get_current_vdac_setting(),mod);
 	return 0;
 }
diff --git a/drivers/amlogic/display/vout/tvconf.h b/drivers/amlogic/display/vout/tvconf.h
index 4f066c50adff..cb91eee43f90 100755
--- a/drivers/amlogic/display/vout/tvconf.h
+++ b/drivers/amlogic/display/vout/tvconf.h
@@ -56,4 +56,6 @@ __ATTR(name, S_IRUGO|S_IWUSR, aml_TV_attr_##name##_show, aml_TV_attr_##name##_st
 **
 ******************************************************************/
 static void  parse_vdac_setting(char *para);
+tvmode_t vmode_to_tvmode(vmode_t mod);
+
 #endif
diff --git a/drivers/amlogic/display/vout/tvoutc.c b/drivers/amlogic/display/vout/tvoutc.c
index 809a4549bfbd..7960cbefa678 100755
--- a/drivers/amlogic/display/vout/tvoutc.c
+++ b/drivers/amlogic/display/vout/tvoutc.c
@@ -29,7 +29,9 @@
 #include <mach/am_regs.h>
 
 #include <linux/amlogic/vout/vinfo.h>
+#include <linux/amlogic/logo/logo.h>
 #include "tvoutc.h"
+#include "tvconf.h"
 #include <linux/clk.h>
 #include <plat/io.h>
 #include <mach/tvregs.h>
@@ -253,6 +255,12 @@ static void set_tvmode_misc(tvmode_t mode)
  */
 static int uboot_display_already(tvmode_t mode)
 {
+    tvmode_t source = vmode_to_tvmode(get_resolution_vmode());
+    if(source == mode)
+        return 1;
+    else 
+        return 0;
+    /*
     const  reg_t *s = tvregsTab[mode];
     unsigned int pxcnt_tab = 0;
     unsigned int lncnt_tab = 0;
@@ -273,6 +281,7 @@ static int uboot_display_already(tvmode_t mode)
     } else {
         return 0;
     }
+    */
 }
 
 int tvoutc_setmode(tvmode_t mode)
diff --git a/drivers/amlogic/hdmi/hdmi_tx/hdmi_tx.c b/drivers/amlogic/hdmi/hdmi_tx/hdmi_tx.c
index f01253fea716..74e1c9ec8181 100755
--- a/drivers/amlogic/hdmi/hdmi_tx/hdmi_tx.c
+++ b/drivers/amlogic/hdmi/hdmi_tx/hdmi_tx.c
@@ -1175,9 +1175,11 @@ edid_op:
             hdmitx_device->HWOp.CntlConfig(hdmitx_device, CONF_CLR_AVI_PACKET, 0);
             hdmitx_device->HWOp.CntlConfig(hdmitx_device, CONF_CLR_VSDB_PACKET, 0);
             // if VIID PLL is using, then disable HPLL
+            #if 0
             if(hdmitx_device->HWOp.CntlMisc(hdmitx_device, MISC_VIID_IS_USING, 0)) {
                 hdmitx_device->HWOp.CntlMisc(hdmitx_device, MISC_HPLL_OP, HPLL_DISABLE);
             }
+            #endif
             hdmitx_device->HWOp.CntlDDC(hdmitx_device, DDC_HDCP_OP, HDCP_OFF);
             hdmitx_device->HWOp.CntlMisc(hdmitx_device, MISC_TMDS_PHY_OP, TMDS_PHY_DISABLE);
             hdmitx_device->HWOp.Cntl(hdmitx_device, HDMITX_IP_SW_RST, TX_SYS_SW_RST);
@@ -1723,8 +1725,8 @@ static void __exit amhdmitx_exit(void)
     return ;
 }
 
-module_init(amhdmitx_init);
-//arch_initcall(amhdmitx_init);
+//module_init(amhdmitx_init);
+arch_initcall(amhdmitx_init);
 module_exit(amhdmitx_exit);
 
 MODULE_DESCRIPTION("AMLOGIC HDMI TX driver");
-- 
2.19.0

